### ISSUE TITLE
Going to mess up everything.

### ISSUE THREAD

**Issue by vadis365** [GenDeathrow/Defensive-Measures#1]
I'd like to re-organise stuff a little before we start adding code.
The registrations for blocks and items are all using the old deprecated methods :P 
It's just a matter of preference really - but I'm a stubborn bastard ;)

**Comment by GenDeathrow** [comment]
Too be honest if you have newer methods go for it. I picked that up from funwayguy when I was learning 1.9 & 1.10. Ive started modeling and coding for landmines on my side.

**Comment by vadis365** [comment]
There is nothing wrong with what you have it's all good - like I said it's just personal preference.

**Comment by vadis365** [comment]
I'll get to work on a basic turret model if you like?
I was thinking we could have a few different types like:
Basic Arrow Turrets that could be upgraded for rapid fire etc.
Gun Turrets with more Damage etc.
Flame Turrets with flame throwers.
Rocket Launcher Turrets with upgrades for homing missiles.
Laser Turret - because, well, why not?
What do you think?

**Comment by GenDeathrow** [comment]
Yes all ideas I was thinkin, of scary we think a like. .. I had even thought of the laser but thought it may suck adding the beam. I added a project page to the github tables yo just post ideas and make surewe dont cross work on the sam things.  for turrents will you be doing a TE & Entity combo? Entity would be easy for all the targeting stuff.. TE would act like a turrents base for storing items interacting with pipes and possible RF... basically just an interaction block. It doesnt need to tick just store Information? What was your idea on that?

**Comment by GenDeathrow** [comment]
Plus the entity allows mobs to target it .. thwrapy be a must. I kinda want all traps to take damage in some form or fashion. I want the mod to have more interaction than placing the block down and walking away. But not too much micromanaging.

**Comment by vadis365** [comment]
Yeah, I was thinking the base would be a TE that would interact as an inventory - for automation and stuff like feeding in ammo etc. Once placed I suppose it could spawn an Entity on top that does all the other stuff. Originally I was going to just make it a TE as the targeting isn't that hard to do - but if you want it to be able to be attacked by mobs then it would probably have to be a living entity.
I'll work on making the crossbow turret 1st as a proof of concept and we can see where to go from there.

**Comment by GenDeathrow** [comment]
Sounds good. Question do you have Discord or Curse Chat? I have a page on curse chat set up for my Dev work.

**Comment by vadis365** [comment]
Sorry I don't :(
I should really sort something out but I'm to lazy to do it.

**Comment by GenDeathrow** [comment]
lol well if you do get Curse chat set up.. I'll send you an invite.

**Comment by GenDeathrow** [comment]
I'm creating a small landmine, that when walked over creates activates the landmine, and when the player leaves the collision block the land mine explodes. I'm thinking of making land mine shoot into the air and explode in their face. just an idea atm. first get the base code going.

Edit: I will have it set up so we can create different types of land mines.

**Comment by vadis365** [comment]
Got the 1st proof of the basic crossbow turret model
http://imgur.com/a/OikDr

**Comment by GenDeathrow** [comment]
ok... 1st Thats amazing.. second What do you use to model with? You models always look great but I haven't found an easy to use modeling tool. I feel my models may be inferior to that.

**Comment by vadis365** [comment]
That was just bashed together in a old version of techne I have - nothing fancy

**Comment by GenDeathrow** [comment]
Still you got mad skills with laying out models. I have two versions of the landmine I want to use.. a blocky version thats a simple json model and [THIS](http://imgur.com/a/mMSUX) < That would have to be an OBJ model. I cant seem to get my other model loader I used to open up so I cant show you the image.  Trying to figgure out if I want to use OBJ or not. but I do like the look. even if its a little Rounded.

**Comment by GenDeathrow** [comment]
Also I am going to try and get some code done this weekend, but I have got a few things that have some fast approaching deadlines. So in case I disappear for a bit that is why.

**Comment by vadis365** [comment]
No worries - I don't think I will be adding anything this weekend due to to other commitments - but I'll work on some ideas and models if I get time.
